File: Python/lukas/control1.py
# Mi codigo (colli)

correo = input("Ingrese el correo a crear:\n")

pA = correo.index('@')
pP = correo.index('.')

ussr = correo[:pA]
dom = correo[pA+1:pP]
subdm = correo[pP+1:]

if (ussr.isalpha() and dom == 'util') and (subdm == 'cl' or subdm == 'ar' or subdm == 'br' or subdm == 'co' or subdm == "uy"):
    print("Primer check valido")

else :
    if not ussr.isalpha():
        print("- Solo Leyras")
    if dom != 'util':
        print("- Dominio Malo")
    if subdm != "cl" or subdm != 'co' or subdm != 'ar' or subdm != 'uy' or subdm != 'br': #y siguw
        print("- Sub dom malo")

File: Python/lukas/control1.cpp
#include <bits/stdc++.h>
#include <cctype>
using namespace std;

typedef long long ll;
typedef string str;
#define vec vector

int main() {
    str ussr;
    cout << "Ingrese el correo a crear:" << endl;
    cin >> ussr;
    
    str usr = ussr.substr(0, ussr.find('@'));
    str dom = ussr.substr(ussr.find('@') + 1, ussr.find('.') - ussr.find('@') - 1);
    str subdom = ussr.substr(ussr.find('.') + 1);

    bool letras = 1;

    for (char p : usr) {
        if (!isalpha(p)) letras = 0;
    }

    vec <str> sub = {"cl", "ar", "co", "br", "uy"};
    auto it = find(sub.begin(), sub.end(),subdom);

    if (letras && dom == "util" && it != sub.end() ) {
        cout << "Paso el primer check" << endl;
    }
    else {
        cout << "Su correo presenta los siguinetes errores:" << endl;
        if (!letras) printf("- Solo use Letras\n");
        if (dom != "util") printf("- Dominio malo, use @util.<subdominio>\n");
        if (it == sub.end()) printf("- SubDominio malo, use .cl, .br, .ar, .uy o .co\n");
    }


    return 0;
}
